projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,313 @@
+package projects.opentrain.gtfs_pipeline.heatmap;
+
+import com.projects.opentrain.gtfs_pipeline.Protos.Stop;
+import com.projects.opentrain.gtfs_pipeline.Protos.StopTime;
+import com.projects.opentrain.gtfs_pipeline.Protos.Trip;
+import com.google.startupos.common.flags.Flags;
+
+import javax.imageio.ImageIO;
+
+import java.awt.Graphics2D;
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.startupos.common.flags.Flag;
+import com.google.startupos.common.flags.FlagDesc;
+
+public class HeatMap {
+
+  public List<Stops_around> stopsAround = new ArrayList<>();
+
+  public int[][] heatMap;
+
+  @FlagDesc(
+      name = "image_path",
+      description = "Image path to create heatmap image,need to contain .../imageName.png")
+  public static Flag<String> imagePath = Flag.create("");
+
+  public class Stops_around // the stop and his location on arrival_time matrix
+   {
+
+    int i = 0; // index on heatMap
+    int j = 0;
+    Stop stop;
+    double arrivalTime;
+
+    public Stops_around(StopTime stopT, double arrivalT) {
+
+      arrivalTime = arrivalT;
+      Stop s;
+      FileInputStream StopInput = null;
+
+      try {
+        StopInput =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/stops.protobin"));
+      } catch (FileNotFoundException e) {
+        e.printStackTrace();
+      }
+
+      while (true) { // from all stops class pick my stop from stopsTime *to use lon/lat
+        try {
+          s = Stop.parseDelimitedFrom(StopInput);
+
+          if (s == null) break;
+          if (s.getStopId() == stopT.getStopId()) {
+            stop = s;
+            StopInput.close();
+            break;
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+  }
+
+  public void arrival_time_to_location() { // by stops.lat,stops.lon make matrix of colors
+
+    double max_lon = 0; // longitude
```

Comment:
I don't see a difference because i don't use it so much,
why its better without a comment?

Reply:
I don't suggest removing all the comments in the code. Sometimes we can't avoid it. But if you can simplify/rename something to avoid comments please do it. It's better to understand what means variable/method/class just looking at the name than looking for comments for it. Sometimes code is changing but comments remain old and unactual. It can confuse.